cli/azd/pkg/apphost/generate.go

```diff
@@ -369,6 +430,58 @@ func (b *infraGenerator) LoadManifest(m *Manifest) error {
 				// a part of a server and it should not be created as a separate resource.
 				b.addContainerAppService(name, "postgres")
 			}
+		case "parameter.v0":
+			b.addInputParameter(name, comp.Inputs["value"].Type)
+		case "azure.bicep.v0":
+			if comp.Path == nil {
+				if comp.Parent == nil {
+					return fmt.Errorf("bicep resource %s does not have a path or a parent", name)
+				}
+				// module uses parent
+				continue
+			}
+			if comp.Params == nil {
+				comp.Params = make(map[string]any)
+			}
+			useSecretOutputs := false
+			if comp.ConnectionString != nil && strings.Contains(*comp.ConnectionString, "secretOutputs") {
+				useSecretOutputs = true
+			}
+			keyVaultForParam := ""
+			emptyJsonString := "\"\""
+			for p, pVal := range comp.Params {
+				jsonBytes, err := json.Marshal(pVal)
+				finalParamValue := string(jsonBytes)
+				if err != nil {
+					return fmt.Errorf("marshalling param %s. error: %w", p, err)
+				}
+				if p == "keyVaultName" && useSecretOutputs {
+					keyVaultForParam = name + "kv"
+					keyVaultParam := fmt.Sprintf(
+						"resources.outputs.SERVICE_BINDING_%s_NAME", strings.ToUpper(keyVaultForParam))
+					if finalParamValue == emptyJsonString {
+						finalParamValue = keyVaultParam
+					}
+				}
+				if p == "principalId" && finalParamValue == emptyJsonString {
+					finalParamValue = "guid(resources.outputs.AZURE_CONTAINER_REGISTRY_MANAGED_IDENTITY_ID)"
```

Comment:
Slightly confused about this part - the `guid` function in bicep create a GUID based on the inputs (I assume it a V4 style UUID where the "randomness" is "we hashed all the inputs and used the hash result as the entropy).  But it seems like the thing this going to yield isn't going to be what the module expects.  I suspect this needs to be the actual object ID of some managed identity so it can create a role assignment?

It would be nice for us to write down somewhere in prose the set of bicep inputs that `azd` understands - how it knows about them (from this it looks like it is just specific names, which is fine) and what the expectation of the value that `azd` will supply will be.

Reply:
I think this needs to get fixed before we check this in, looking at the generated bicep - it seems like what we are doing will end up creating a role assignment for a bogus principal?  I think this should be just `resources.outputs.AZURE_CONTAINER_REGISTRY_MANAGED_IDENTITY_ID`

Reply:
using `resources.outputs.AZURE_CONTAINER_REGISTRY_MANAGED_IDENTITY_ID` fails.
That was my first attempt.  The response from the appContainer was that I needed a guid, not a resourceId. 
Then I saw that `gui()` in bicep can resolve a resourceId into its guid